src/controller/python/chip/clusters/Attribute.py

```diff
@@ -745,18 +788,21 @@ def _handleReportBegin(self):
         pass
 
     def _handleReportEnd(self):
-        self._cache.UpdateCachedData(self._changedPathSet)
-
         if (self._subscription_handler is not None):
             for change in self._changedPathSet:
-                try:
-                    attribute_path = TypedAttributePath(Path=change)
-                except (KeyError, ValueError) as err:
-                    # path could not be resolved into a TypedAttributePath
-                    LOGGER.exception(err)
-                    continue
-                self._subscription_handler.OnAttributeChangeCb(
-                    attribute_path, self._subscription_handler)
+                if self._subscription_handler.OnAttributeChangeCb:
```

Comment:
Maybe change this to a guard to make the code less nested ?
So continue in this for loop if there is no 'self._subscription_handler.OnAttributeChangeCb'

Reply:
Yeah same here, that change came also from the raw attribute access change I've dropped from this PR again. Sorry about that.